lib/scholar/metrics/regression.ex

```diff
@@ -519,6 +519,118 @@ defmodule Scholar.Metrics.Regression do
     Nx.reduce_max(Nx.abs(y_true - y_pred))
   end
 
+  mean_pinball_loss_opts = [
+    alpha: [
+      type: :float,
+      default: 0.5,
+      doc: """
+      The slope of the pinball loss, default=0.5,
+      This loss is equivalent to $$mean_absolute_error$$ when $$\alpha$$ is 0.5,
+      $$\alpha = 0.95$$ is minimized by estimators of the 95th percentile.
+      """
+    ],
+    sample_weights: [
+      type:
+        {:or,
+         [
+           {:custom, Scholar.Options, :weights, []},
+           {:custom, Scholar.Options, :multi_weights, []}           
+         ]},
+      doc: """
+      The weights for each observation. If not provided,
+      all observations are assigned equal weight.
+      """
+    ],
+    multioutput: [
+      type: {:or,
+             [
+               {:custom, Scholar.Options, :weights, []},
+               {:in, [:raw_values, :uniform_average]}
+             ]
+      },
+      default: :uniform_average,
+      doc: """
+      Defines aggregating of multiple output values.
+      Array-like value defines weights used to average errors.
+      Defaults to `:uniform_average`.
+
+        `:raw_values` :
+            Returns a full set of errors in case of multioutput input.
+
+        `:uniform_average` :
+            Errors of all outputs are averaged with uniform weight.
+      
+      The weights for each observation. If not provided,
+      all observations are assigned equal weight.
+      """      
+    ]
+  ]
+  @mean_pinball_loss_schema NimbleOptions.new!(mean_pinball_loss_opts)
+  
+  @doc ~S"""
+  Calculates the mean pinball loss to evaluate predictive performance of quantile regression models.
+
+  $$pinball(y, \hat{y}) = \frac{1}{n) \sum_{i=1}^{n} \alpha max(\hat{y_i} - y_i, 0) +
+  (1 - \alpha) max(\hat{y_i} - y_i, 0)$$
+
+  The residual error is defined as $$|y - \hat{y}|$$ where $y$ is a true value
+  and $\hat{y}$ is a predicted value.
+
+  #{NimbleOptions.docs(@mean_pinball_loss_schema)}
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([1, 2, 3])
+      iex> y_pred = Nx.tensor([2, 3, 4])
+      iex> Scholar.Metrics.Regression.mean_pinball_loss(y_true, y_pred)
+      #Nx.Tensor<
+        f32
+        0.5
+      >
+      iex> y_true = Nx.tensor([[1, 0, 0, 1], [0, 1, 1, 1], [1, 1, 0, 1]])
+      iex> y_pred = Nx.tensor([[0, 0, 0, 1], [1, 0, 1, 1], [0, 0, 0, 1]])
+      iex> Scholar.Metrics.Regression.mean_pinball_loss(y_true, y_pred, alpha: 0.5, multioutput: :raw_values)
+      #Nx.Tensor<
+        f32[4]
+        [0.5, 0.3333333432674408, 0.0, 0.0]
+      >
+  """
+  deftransform mean_pinball_loss(y_true, y_pred, opts \\ []) do
+    mean_pinball_loss_n(y_true, y_pred, NimbleOptions.validate!(opts, @mean_pinball_loss_schema))
+  end
+  defnp mean_pinball_loss_n(y_true, y_pred, opts \\ []) do
```

Comment:
In defnp, opts are always passed (otherwise there should not be opts ;) ) 
```suggestion
  defnp mean_pinball_loss_n(y_true, y_pred, opts) do
```